gfp12_test.go

```diff
@@ -0,0 +1,84 @@
+package bn256
+
+import (
+	"math/big"
+	"testing"
+)
+
+func TestGfP12SpecialSquare(t *testing.T) {
+	in := gfP12Gen
+
+	got := &gfP12{}
+	expected := &gfP12{}
+
+	got.SpecialSquare(in)

```

Comment:
add some warning, since this test assume the input is in the cyclotomic subgroup.

Reply:
added requirement for input